driver-core/src/main/java/com/datastax/driver/core/Metadata.java

```diff
@@ -439,7 +455,7 @@ static String escapeId(String ident) {
      * or even {@link Cluster#connect(String)}.
      */
     public static String quote(String id) {
-        return '"' + id + '"';
+        return '"' + id.replace("\"", "\"\"") + '"';
```

Comment:
Also backported from 3.0; I don't understand why this wasn't already present in 2.1 as without this replacement any identifier containing quotes will be wrongly quoted.